lib/ddq-backend-mysql.js

```diff
@@ -0,0 +1,386 @@
+        ddqBackendInstance.connection.query(`UPDATE ??
+            SET heartbeatDate = NOW()
+            WHERE hash = ?
+            AND isProcessing = true
+            AND owner = ?;`,
+            [
+                ddqBackendInstance.config.table,
+                recordId,
+                ddqBackendInstance.owner
+            ],
+            (err) => {
+                if (err) {
+                    console.error(err);
+                }
+
+                callback(err);
+            }
+        );
+    }
+
+
+    /**
+     * Polls for a random record with the minimum delay defined by the
+     * config.pollDelayMs.
+     *
+     * @param {Object} ddqBackendInstance
+     * @param {Function} callback
+     */
+    function poll(ddqBackendInstance, callback) {
+        /**
+         * Polls for a random record in the database and calls the callback
+         * provided to it. This will continue until pausePolling is called.
+         */
+        function findData() {
+            ddqBackendInstance.connection.query(`SELECT hash
+                FROM ??
+                WHERE isProcessing = false
+                AND topics IN (?)
+                OR topics IS NULL
+                ORDER BY RAND()
+                LIMIT 1;`,
+                [
+                    ddqBackendInstance.config.table,
+                    ddqBackendInstance.config.topics
+                ],
+                callback
+            );
+
+            ddqBackendInstance.poller = timers.setTimeout(findData, ddqBackendInstance.config.pollDelayMs);
+        }
+
+        ddqBackendInstance.poller = timers.setTimeout(findData, ddqBackendInstance.config.pollDelayMs);
+    }
+
+
+    /**
+     * Updates a specific record so that it can be requeued.
+     *
+     * @param {Object} ddqBackendInstance
+     * @param {string} recordId
+     * @param {Function} callback
+     */
+    function requeue(ddqBackendInstance, recordId, callback) {
+        ddqBackendInstance.connection.query(`UPDATE ??
+            SET owner = null, isProcessing = false, requeued = true
+            WHERE hash = ?;`,
+            [
+                ddqBackendInstance.config.table,
+                recordId
+            ],
+            callback
+        );
+    }
+
+
+    /**
+     * Deletes a specific record from the database. If unsuccessful, this
+     * will call the command to requeue the record.
+     *
+     * @param {Object} ddqBackendInstance
+     * @param {string} recordId
+     */
+    function remove(ddqBackendInstance, recordId) {
+        ddqBackendInstance.connection.query(`DELETE FROM ??
+            WHERE hash = ?
+            AND requeued = false;`,
+            [
+                ddqBackendInstance.config.table,
+                recordId
+            ],
+            (err) => {
+                if (err) {
+                    console.error("Error deleting record with id:", recordId);
+                    console.error("Attempting requeue");
```

Comment:
Removed all logging. If we decide that debug logging would be good to have, I'll create a separate feature branch for it later on.